src/core/obj.js

```diff
@@ -315,6 +315,67 @@ class Catalog {
       return onParsed;
     }
 
+    function parseOrder(refs, nestedLevels = 0) {
```

Comment:
The first object in a subarray of the `Order` entry can be also an OCG instead of a string. The OCG becomes an expandable node. If it is set to invisible, all child nodes should get grayed out. See these files:
[PDFill_layers.pdf](https://github.com/mozilla/pdf.js/files/5052227/PDFill_layers.pdf)
[pdflib_hierarchical_layers.pdf](https://github.com/mozilla/pdf.js/files/5052228/pdflib_hierarchical_layers.pdf)
Also the mutually exclusive feature of radio buttons (`RBGroups`) and locked OCGs seems to be still unimplemented, if I'm right?

Reply:
> The first object in a subarray of the `Order` entry can be also an OCG instead of a string. The OCG becomes an expandable node.

Very interesting, I hadn't seen those types of PDF documents when writing the patches.
I'll try to see how difficult implementing that would be; or if we could possibly defer that for a follow-up PR.

> Also the mutually exclusive feature of radio buttons (`RBGroups`) and locked OCGs seems to be still unimplemented, if I'm right?

Yes, those weren't implemented in PR #12095 and here I simply implemented the *bare minimum* of additional "core" functionality to get the viewer integration to work OK.
I'd really prefer if those things were implemented separately though, since this PR is already quite large (and those features also don't seem *that* common); there's also the question if we'd even want to respect the "locked OCGs" in the viewer.

Reply:
> > The first object in a subarray of the `Order` entry can be also an OCG instead of a string. The OCG becomes an expandable node.
> 
> Very interesting, I hadn't seen those types of PDF documents when writing the patches.
> I'll try to see how difficult implementing that would be; or if we could possibly defer that for a follow-up PR.

Having looking (quickly) at this, I'd *really* prefer if we could defer that to a follow-up patch instead.